web/src/ShareSnapshotModal.tsx

```diff
@@ -215,3 +243,81 @@ export default class ShareSnapshotModal extends PureComponent<props> {
     })
   }
 }
+
+type DownloadSnapshotModalProps = {
+  handleClose: () => void
+  getSnapshot: () => Proto.webviewSnapshot
+  isOpen: boolean
+  dialogAnchor: HTMLElement | null
+}
+
+const SaveSnapshotButton = styled(InstrumentedButton)`
+  background-color: ${Color.gray70};
+  border: 1px solid ${Color.gray10};
+  color: ${Color.gray10};
+  display: flex;
+  font-family: ${Font.monospace};
+  font-size: ${FontSize.default};
+  margin: ${SizeUnit(0.75)} 0;
+  text-transform: unset;
+
+  &:focus,
+  &:active {
+    background-color: rgba(0, 0, 0, 0.04);
+  }
+`
+
+const CodeSnippet = styled.code`
+  background-color: ${Color.offWhite};
+  padding: 0 5px;
+  white-space: nowrap;
+`
+
+function downloadSnapshot(snapshot: Proto.webviewSnapshot) {
+  const data = new Blob([JSON.stringify(snapshot)], {
+    type: "application/json",
+  })
+  saveAs(data, "snapshot.json")
```

Comment:
small nit: it might be good to put the current datetime in the file name, similar to how OS screenshotting tools do it, so you don't have snapshots overwriting each other